tools/Propulsion.Tool/Program.fs

```diff
@@ -24,14 +24,30 @@ type Parameters =
 
 and [<NoComparison; NoEquality>] InitAuxParameters =
     | [<AltCommandLine("-ru"); Mandatory>]  Rus of int
+    | [<AltCommandLine "-A">]               Autoscale
+    | [<AltCommandLine "-m">]               Mode of CosmosModeType
     | [<AltCommandLine("-s")>]              Suffix of string
     | [<CliPrefix(CliPrefix.None)>]         Cosmos of ParseResults<Args.Cosmos.Parameters>
     interface IArgParserTemplate with
         member a.Usage = a |> function
             | Rus _ ->                      "Specify RU/s level to provision for the Aux Container."
+            | Autoscale ->                  "Autoscale provisioned throughput. Use --rus to specify the maximum RU/s."
+            | Mode _ ->                     "Configure RU mode to use Container-level RU, Database-level RU, or Serverless allocations (Default: Use Container-level allocation)."
             | Suffix _ ->                   "Specify Container Name suffix (default: `-aux`)."
             | Cosmos _ ->                   "Cosmos Connection parameters."
 
+and CosmosInitInfo(args : ParseResults<InitAuxParameters>) =
+    member _.ProvisioningMode =
+        let throughput () =
```

Comment:
maybe extract this as `member private _.ThroughputSpec ?`

Reply:
I have used a let binding instead of a member. My understanding is that a function execution can be avoided. Please correct me if I'm wrong.

Reply:
My main intent was legibility, though we are clearly on the same page with trying to avoid invocations, which is probably why it was this way! (of course perf is irrelevant as we are paying lots of ms for reflection so we can have readable code vs actually optimizing startup time when using Argu!)

in terms of semantics, a `member private` would be equivalent as its a property getter function which would be invoked just in time (`member val` would init a temp in the ctor phase)